mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java

```diff
@@ -0,0 +1,283 @@
+                .and(set(SEEN, bindMarker(SEEN)))
+                .and(set(USER, bindMarker(USER)))
+                .and(set(USER_FLAGS, bindMarker(USER_FLAGS)))
+                .and(set(MOD_SEQ, bindMarker(MOD_SEQ)))
+                .where(eq(MESSAGE_ID, bindMarker(MESSAGE_ID)))
+                .onlyIf(eq(MOD_SEQ, bindMarker(MOD_SEQ))));
+    }
+
+    private PreparedStatement deleteStatement(Session session) {
+        return session.prepare(QueryBuilder.delete()
+                .from(TABLE_NAME)
+                .where(eq(MESSAGE_ID, bindMarker(MESSAGE_ID))));
+    }
+
+    public CompletableFuture<Void> save(Mailbox mailbox, MailboxMessage message) throws MailboxException {
+        try {
+            CassandraMessageId messageId = (CassandraMessageId) message.getMessageId();
+            BoundStatement boundStatement = insertStatement.bind()
+                .setUUID(MESSAGE_ID, messageId.get())
+                .setLong(MOD_SEQ, message.getModSeq())
+                .setDate(INTERNAL_DATE, message.getInternalDate())
+                .setInt(BODY_START_OCTET, (int) (message.getFullContentOctets() - message.getBodyOctets()))
+                .setLong(FULL_CONTENT_OCTETS, message.getFullContentOctets())
+                .setLong(BODY_OCTECTS, message.getBodyOctets())
+                .setBool(ANSWERED, message.isAnswered())
+                .setBool(DELETED, message.isDeleted())
+                .setBool(DRAFT, message.isDraft())
+                .setBool(FLAGGED, message.isFlagged())
+                .setBool(RECENT, message.isRecent())
+                .setBool(SEEN, message.isSeen())
+                .setBool(USER, message.createFlags().contains(Flag.USER))
+                .setSet(USER_FLAGS, userFlagsSet(message))
+                .setBytes(BODY_CONTENT, toByteBuffer(message.getBodyContent()))
+                .setBytes(HEADER_CONTENT, toByteBuffer(message.getHeaderContent()))
+                .setList(PROPERTIES, message.getProperties().stream()
+                    .map(x -> typesProvider.getDefinedUserType(PROPERTIES)
+                        .newValue()
+                        .setString(Properties.NAMESPACE, x.getNamespace())
+                        .setString(Properties.NAME, x.getLocalName())
+                        .setString(Properties.VALUE, x.getValue()))
+                    .collect(Collectors.toList()))
+                .setList(ATTACHMENTS, message.getAttachments().stream()
+                    .map(this::toUDT)
+                    .collect(Collectors.toList()));
+
+            setTextualLineCount(boundStatement, message.getTextualLineCount());
+
+            return cassandraAsyncExecutor.executeVoid(boundStatement);
+
+        } catch (IOException e) {
+            throw new MailboxException("Error saving mail", e);
+        }
+    }
+
+    private void setTextualLineCount(BoundStatement boundStatement, Long textualLineCount) {
+        if (textualLineCount == null) {
```

Comment:
Not as simple, as the orElse part will be interpreted everytime; the TEXTUAL_LINE_COUNT will always be null...